specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/Patches.json

```diff
@@ -0,0 +1,392 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ContainerApps API Client",
+    "version": "2023-05-02-preview"
+  },
+  "schemes": [
+    "https"
+  ],
+  "host": "management.azure.com",
+  "produces": [
+    "application/json"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "flow": "implicit",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/builders/{builderName}/patches": {
+      "get": {
+        "tags": [
+          "Patches"
+        ],
+        "operationId": "Patches_ListByBuilderResource",
+        "description": "List PatchResource resources by BuilderResource.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./Builders.json#/parameters/BuilderNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "ARM operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/PatchCollection"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Patches_Apply_0": {
+            "$ref": "./examples/Patches_ListByBuilderResource.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/builders/{builderName}/patches/{patchName}": {
+      "get": {
+        "tags": [
+          "Patches"
+        ],
+        "operationId": "Patches_Get",
+        "description": "Get a list of patches to be applied.",
```

Comment:
Can you explain to me how we will have multiple patches returned under a single `patchName` GET request? Generally, the GET request with this route will return a single `patch` resource, rather than a list

Reply:
Sure! This operation will be doing DB retrieval and get the specific patches that ties to the cluster this builder resource is in (and also those that are tied to this patcher). This GET request is not returning the patch resource, but to return the patches related to this specific patcher with the name `{patcherName}` specified in the id